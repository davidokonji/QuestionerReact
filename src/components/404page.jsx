import React, { Fragment } from 'react';
import Header from './Landingheader';

const NotFound = (props) => {
  const { text } = props;
  return (
    <Fragment>
      <Header
        page="landingpage"
        navClass="nav_header"
      />
      <div id="notfound">
        <div className="notfound">
          <div className="notfound-404">
            <h3>
            Oops!
              {
              text || ' Page '
            }
              not found

            </h3>
            <h1>
              <span>4</span>
              <span>0</span>
              <span>4</span>
            </h1>
          </div>
          <h2>
          we are sorry, page you requested was not found
          </h2>
          <a href="/" className="homeLink">home</a>
        </div>
      </div>
    </Fragment>
  );
};

export default NotFound;
