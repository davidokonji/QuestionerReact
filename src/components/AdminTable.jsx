import React from 'react';
import MeetupList from './MeetupList';

const AdminTable = ({ data, toggle }) => (

  // const deleteMeeteps = ( id) => {
  //     this.props.delete(id)
  // }
  <div className='col-11 mx-4 mx-md-0 mt-4 mt-md-0 col-md-9 shadow p-0 table-responsive-sm'>
    <table className='table table-hover p-0'>
      <thead className='row mx-0'>
        <div className='col-10 bg-light p-3'>
            Available Meetup(s)
        </div>
        <div className='col-2 bg-light p-3'>
            Operation
        </div>
      </thead>
      <tbody className='row p-2 col-12 ml-0 '>
        {
          data.map(meetup => (
            <MeetupList
              id={meetup.id}
              title={meetup.title}
              location={meetup.location}
              tags={meetup.tags}
              happeningOn={meetup.happeningOn}
              images={meetup.images}
              key={meetup.id}
              // delete={delteMeetups}
            //   deleteDialog={toggle}
            />
          ))
      }
      </tbody>
    </table>
  </div>
);

export default AdminTable;
