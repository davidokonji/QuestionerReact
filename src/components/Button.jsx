import React from 'react';

const Button = (props) => {
  const { text, event, styles } = props;
  return (
    <button type="button" onClick={event} className={styles}>
      {text}
    </button>
  );
};

export default Button;
