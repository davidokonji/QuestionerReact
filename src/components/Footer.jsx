import React from 'react';
import '../style/footer.scss';

const Footer = () => {
  const date = new Date().getFullYear();
  return (
    <React.Fragment>
      <footer>
      &copy;
        {date}

      </footer>
    </React.Fragment>
  );
};

export default Footer;
