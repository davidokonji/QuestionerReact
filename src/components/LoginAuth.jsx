import React, { Component } from 'react';
import { Redirect, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import { LoginAction } from '../actions';

export class LoginComponent extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      message: '',
      disable: false,
    };
  }

  handleInputChange = (event) => {
    const { target } = event;
    const { value } = target;
    const { name } = target;
    this.setState({
      [name]: value,
    });
  }

  handleSignin = async (e) => {
    e.preventDefault();
    const { email, password } = this.state;
    const { userLogin } = this.props;
    const data = {
      email,
      password
    };
    await userLogin(data);
  }


  render() {
    const {
      email,
      password,
      message,
      disable
    } = this.state;
    const { user = '', redirect = false, history: { push } } = this.props;
    if (redirect) {
      toast(user.message, {
        position: toast.POSITION.TOP_CENTER,
        type: toast.TYPE.SUCCESS,
        onClose: () => push('/dashboard')
      });
    }
    return (
      <React.Fragment>
        <form method='post' onSubmit={this.handleSignin}>
          <div className='login-form' id='login-form'>
            <input
              type='text'
              name='email'
              value={email}
              onChange={this.handleInputChange}
              placeholder='Email'
              className='input form_lg'
              required
            />
            <input
              type='password'
              name='password'
              value={password}
              onChange={this.handleInputChange}
              placeholder='Password'
              id='password'
              className='input form_lg'
              required
            />
            <div>
              <button
                className='btn'
                id='login'
                type='submit'
                disabled={disable}
              >
              Sign in
              </button>
            </div>
          </div>
        </form>
      </React.Fragment>
    );
  }
}

export const mapStateToProps = state => ({
  user: state.auth,
  redirect: state.auth.redirect
});

const mapDispatchToProps = {
  userLogin: LoginAction
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(LoginComponent));
