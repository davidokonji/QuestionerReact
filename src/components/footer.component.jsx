import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const Footer = () => (
  <footer className="footer">
    <div className="footerBottom">
      <p>
       Created with
        {<FontAwesomeIcon icon="heart" size="sm" color="red" style={{ paddingLeft: '1rem', paddingRight: '1rem' }} />}
       by Okonji Chukwunonso
      </p>
    </div>
  </footer>
);

export default Footer;
