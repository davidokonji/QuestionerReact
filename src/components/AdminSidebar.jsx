import React, { Fragment, Component } from 'react';
import { connect } from 'react-redux';
import { WithContext as ReactTags } from 'react-tag-input';
import {
  Form,
  FormGroup,
  Label,
  Input,
  FormText,
} from 'reactstrap';
import Button from './Button';
import { addMeetup } from '../actions';
import Modal from './common/modal';

const KeyCodes = {
  comma: 188,
  enter: 13,
};

const delimiters = [KeyCodes.comma, KeyCodes.enter];
export class AdminSidebar extends Component {
  constructor(props) {
    super(props);

    this.state = {
      tagList: [
      ],
      suggestions: [
        { id: 'technology', text: 'technology' },
        { id: 'social', text: 'social' },
      ],
      topic: '',
      location: '',
      happeningOn: '',
      images: '',
      imagePreview: ''
    };
  }

  handleOnChange = (e) => {
    const { name, value } = e.target;
    this.setState({
      [name]: value
    });
  }

  imageHandler = ({ target: { files } }) => {
    this.setState({
      images: files[0],
      imagePreview: URL.createObjectURL(files[0]),
    });
  }


  handleSubmit = (e) => {
    e.preventDefault();
    const { createMeetup } = this.props;
    const {
      topic,
      location,
      happeningOn,
      tagList,
      images
    } = this.state;
    const filteredTags = tagList.map(tag => tag.text);
    const tags = filteredTags.join();
    const data = {
      topic,
      location,
      happeningOn,
      tags,
      images
    };

    createMeetup(data);
  }

  handleAddition = (tag) => {
    this.setState(state => ({ tagList: [...state.tagList, tag] }));
  }

  handleDrag = (tag, currPos, newPos) => {
    const { tagList } = this.state;
    const setTags = new Set([...tagList]);
    const newTags = setTags.slice();

    newTags.splice(currPos, 1);
    newTags.splice(newPos, 0, tag);

    this.setState({ tagList: newTags });
  }

  handleDelete = (i) => {
    const { tagList } = this.state;
    this.setState({
      tagList: tagList.filter((tag, index) => index !== i),
    });
  }

  render() {
    const { toggle, open } = this.props;
    const {
      tagList,
      suggestions,
      imagePreview,
      topic
    } = this.state;
    const placeholder = 'Add meetup Tags';
    return (
      <Fragment>
        <div className='col-12 col-md-3'>
          <div className='p-5 shadow-sm'>
            <Button styles='button p-3' text='Create Meetup' event={toggle} />
          </div>
        </div>
        <Modal toggle={toggle} open={open} className='mt-5'>
          <Form method='post' onSubmit={this.handleSubmit} encType='multipart/form-data'>
            {
              imagePreview && (
                <div>
                  <img src={imagePreview} alt={topic} style={{ height: '30rem' }} />
                </div>
              )
            }
            <FormGroup>
              <Input
                type='file'
                name='image'
                max={1}
                accept='image/*'
                onChange={this.imageHandler}
              />
              <FormText color='muted'>
              Add meetup images
              </FormText>
            </FormGroup>
            <FormGroup>
              <Input
                type='text'
                name='topic'
                placeholder='Add Topic'
                onChange={this.handleOnChange}
                className='px-3'
              />
            </FormGroup>
            <FormGroup>
              <Input
                type='text'
                name='location'
                placeholder='Add Location'
                onChange={this.handleOnChange}
              />
            </FormGroup>
            <FormGroup>
              <Input
                type='date'
                name='happeningOn'
                placeholder='Add Date'
                onChange={this.handleOnChange}
              />
            </FormGroup>
            <FormGroup>
              <div id='app'>
                <ReactTags
                  tags={tagList}
                  suggestions={suggestions}
                  handleDelete={this.handleDelete}
                  handleAddition={this.handleAddition}
                  handleDrag={this.handleDrag}
                  delimiters={delimiters}
                  placeholder={placeholder}
                  inline={false}
                />
              </div>
            </FormGroup>
            <button type='submit' className='btn' onClick={toggle}>Create Meetup</button>
          </Form>
        </Modal>
      </Fragment>
    );
  }
}
const mapDispatchToProps = {
  createMeetup: addMeetup
};


export default connect(null, mapDispatchToProps)(AdminSidebar);
