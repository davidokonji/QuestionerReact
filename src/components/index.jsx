import React from 'react';
import Navbar from './Navbar';
import Slider from './slider';
import Footer from './Footer';

const App = () => {
  return (
    <React.Fragment>
      <Navbar />
      <Slider />
      <Footer />
    </React.Fragment>
  );
};

export default App;
