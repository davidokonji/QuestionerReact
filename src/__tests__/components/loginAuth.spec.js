import React from 'react';
import { shallow } from 'enzyme';
import { LoginComponent, mapStateToProps } from '../../components/LoginAuth';

let wrapper;
describe('Login Compnent', () => {
  beforeAll(() => {
    const userLogin = jest.fn();
    const push = jest.fn();
    const history = {
      history: {
        push
      }
    };
    const redirect = {
      redirect: true
    };
    wrapper = shallow(<LoginComponent userLogin={userLogin} history={history} redirect={redirect} />);
  });
  it('should render Login component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle email change', () => {
    const value = 'email@email.com';

    wrapper.find('input').at(0).simulate('change', {
      target: { value }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle password change', () => {
    const value = 'password';

    wrapper.find('input').at(1).simulate('change', {
      target: { value }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle form submit', () => {
    wrapper.find('form').simulate('submit', {
      preventDefault: () => {}
    });

    wrapper.setState({
      error: ''
    });
    expect(wrapper).toMatchSnapshot();
  });

  // it('should test mapStateToProps object', () => {
    
  // });
});
