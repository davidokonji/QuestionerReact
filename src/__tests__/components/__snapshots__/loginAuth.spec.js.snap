// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Compnent should handle email change 1`] = `
<Fragment>
  <form
    method="post"
    onSubmit={[Function]}
  >
    <div
      className="login-form"
      id="login-form"
    >
      <input
        className="input form_lg"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="text"
        value=""
      />
      <input
        className="input form_lg"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
      <div>
        <button
          className="btn"
          disabled={false}
          id="login"
          type="submit"
        >
          Sign in
        </button>
      </div>
    </div>
  </form>
</Fragment>
`;

exports[`Login Compnent should handle form submit 1`] = `
<Fragment>
  <form
    method="post"
    onSubmit={[Function]}
  >
    <div
      className="login-form"
      id="login-form"
    >
      <input
        className="input form_lg"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="text"
        value=""
      />
      <input
        className="input form_lg"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
      <div>
        <button
          className="btn"
          disabled={false}
          id="login"
          type="submit"
        >
          Sign in
        </button>
      </div>
    </div>
  </form>
</Fragment>
`;

exports[`Login Compnent should handle password change 1`] = `
<Fragment>
  <form
    method="post"
    onSubmit={[Function]}
  >
    <div
      className="login-form"
      id="login-form"
    >
      <input
        className="input form_lg"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="text"
        value=""
      />
      <input
        className="input form_lg"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
      <div>
        <button
          className="btn"
          disabled={false}
          id="login"
          type="submit"
        >
          Sign in
        </button>
      </div>
    </div>
  </form>
</Fragment>
`;

exports[`Login Compnent should render Login component correctly 1`] = `
<Fragment>
  <form
    method="post"
    onSubmit={[Function]}
  >
    <div
      className="login-form"
      id="login-form"
    >
      <input
        className="input form_lg"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="text"
        value=""
      />
      <input
        className="input form_lg"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
      <div>
        <button
          className="btn"
          disabled={false}
          id="login"
          type="submit"
        >
          Sign in
        </button>
      </div>
    </div>
  </form>
</Fragment>
`;
